pkg/gateway/model_build_targetgroup.go

```diff
@@ -248,15 +248,19 @@ func (t *latticeServiceModelBuildTask) buildTargetGroup(ctx context.Context, cli
 			if *httpBackendRef.Kind == "Service" {
 				t.Datastore.AddTargetGroup(tgName, "", "", "", tgSpec.Config.IsServiceImport, t.httpRoute.Name)
 			} else {
+				// for serviceimport, the httproutename is ""
 				t.Datastore.AddTargetGroup(tgName, "", "", "", tgSpec.Config.IsServiceImport, "")
 			}
 
 			if t.httpRoute.DeletionTimestamp.IsZero() {
 				// to add
-				t.Datastore.SetTargetGroupByBackendRef(tgName, t.httpRoute.Name, false, true)
+				t.Datastore.SetTargetGroupByBackendRef(tgName, t.httpRoute.Name, tgSpec.Config.IsServiceImport, true)
 			} else {
 				// to delete
-				t.Datastore.SetTargetGroupByBackendRef(tgName, t.httpRoute.Name, false, false)
+				t.Datastore.SetTargetGroupByBackendRef(tgName, t.httpRoute.Name, tgSpec.Config.IsServiceImport, false)
+				dsTG, _ := t.Datastore.GetTargetGroup(tgName, t.httpRoute.Name, tgSpec.Config.IsServiceImport)
+				tgSpec.IsDeleted = true
+				tgSpec.LatticeID = dsTG.ID
```

Comment:
target_group_manger need the `LatticeID` to invoke vpcLattice.deleteTargetgroup Api to handle the target group deletion request 

Change false to tgSpec.Config.IsServiceImport in order to build tg model for both backendref service and ServiceImport, for  ServiceImport, target_group_synthesizer.go will ignore to delete this tg.